connector/web3/web3.go

```diff
@@ -42,19 +56,20 @@ func (c *web3Connector) Verify(address, msg, signedMsg string) (identity connect
 		return identity, fmt.Errorf("could not decode hex string of signed nonce: %v", err)
 	}
 
+	msgHash := signHash([]byte(msg))
+
 	// This is the v parameter in the signature. Per the yellow paper, 27 means even and 28
 	// means odd.
 	if signb[64] == 27 || signb[64] == 28 {
 		signb[64] -= 27
 	} else if signb[64] != 0 && signb[64] != 1 {
 		// We allow 0 or 1 because some non-conformant devices like Ledger use it.
-		// TODO - Verify using ERC-1271
-		return identity, fmt.Errorf("byte at index 64 of signed message should be 27 or 28: %s", signedMsg)
+		return c.VerifyERC1271Signature(addrb, msgHash, signb)
 	}
 
-	pubKey, err := crypto.SigToPub(signHash([]byte(msg)), signb)
+	pubKey, err := crypto.SigToPub(msgHash, signb)
 	if err != nil {
-		return identity, fmt.Errorf("failed to recover public key from signed message: %v", err)
+		return c.VerifyERC1271Signature(addrb, msgHash, signb)
 	}
 
 	recoveredAddr := crypto.PubkeyToAddress(*pubKey)
```

Comment:
I would have expected more action down here. Roughly, you have this old procedure `VerifyEOASignature(address, hash, signature)`. That's roughly lines 61–80 here. If that fails for any reason, you want to try your `VerifyERC1271Signature`.